plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -324,6 +354,80 @@ public CompletableFuture<Collection<Slice>> finish()
         return completedFuture(fragments);
     }
 
+    private Slice writeMergeResult(Slice path, FileDeletion deletion)
+    {
+        Roaring64Bitmap pastDeletionVectors = loadDeletionVectors(Location.of(path.toStringUtf8()));
+
+        Roaring64Bitmap deletedRows = new Roaring64Bitmap();
+        deletedRows.or(pastDeletionVectors);
+        deletedRows.or(deletion.rowsDeletedByDelete());
+        deletedRows.or(deletion.rowsDeletedByUpdate());
+
+        TrinoInputFile inputFile = fileSystem.newInputFile(Location.of(path.toStringUtf8()));
+        try (ParquetDataSource dataSource = new TrinoParquetDataSource(inputFile, parquetReaderOptions, fileFormatDataSourceStats)) {
+            ParquetMetadata parquetMetadata = MetadataReader.readFooter(dataSource, Optional.empty());
+            long rowCount = parquetMetadata.getBlocks().stream().map(BlockMetadata::rowCount).mapToLong(Long::longValue).sum();
+            Roaring64Bitmap deletionVectorForWrite = new Roaring64Bitmap();
+            deletionVectorForWrite.addRange(0, rowCount);
+            deletionVectorForWrite.andNot(deletedRows);
+            if (deletionVectorForWrite.isEmpty()) {
+                // No rows are retained in the file, so we don't need to write deletion vectors.
+                return onlySourceFile(path.toStringUtf8(), deletion);
+            }
+            return writeDeletionVector(path.toStringUtf8(), deletedRows, deletion, parquetMetadata, rowCount);
```

Comment:
Can there be significant overlap between `pastDeletionVectors` and `deletion` ? I'm wondering if we can write a smaller deletion vector through `deletion.andNot(pastDeletionVectors)`